src/scikit_build_core/setuptools/build_cmake.py

```diff
@@ -176,7 +176,7 @@ def _has_cmake(dist: Distribution) -> bool:
     )
 
 
-def _prepare_extension_detection(dist: Distribution) -> None:
+def cmake_extension(dist: Distribution) -> None:
```

Comment:
I am a little confused here, though. Someone doesn't have to specify `cmake_extensions=`, it's not a keyword argument in our API.

However, we do require at least one `cmake_` keyword to be passed. Those could be what ensures this gets called.

(Also, setuptools-rust is a great example and maybe we should check it again to see if matching it's design more closely is a good idea)

We do need to be backward compatible with scikit-build (classic), but we have a wrapper there and we can add whatever we put here there, so that's not really an issue.

Very soon I'm pulling this out into a separate package so it will stop being an issue for modern builds. :)

Reply:
Does setuptools-rust not require a setup.py now?

Reply:
> Does setuptools-rust not require a setup.py now?

I must say I have no idea on that particular.

Reply:
> I am a little confused here, though. Someone doesn't have to specify `cmake_extensions=`, it's not a keyword argument in our API.
> 
> However, we do require at least one `cmake_` keyword to be passed. Those could be what ensures this gets called.
> 

I only know that passing everything into `finalize_distribution_options` leads to breakage because, in the case of sagemath it ends being used in a place it was never meant to be. And I followed setuptools-rust example as to what the appropriate entrypoint for it should be.

> (Also, setuptools-rust is a great example and maybe we should check it again to see if matching it's design more closely is a good idea)
> 
> We do need to be backward compatible with scikit-build (classic), but we have a wrapper there and we can add whatever we put here there, so that's not really an issue.
> 
> Very soon I'm pulling this out into a separate package so it will stop being an issue for modern builds. :)

I suspected that would happen sooner rather than later. This looks like something that should be named setuptools-cmake.

Reply:
> Does setuptools-rust not require a setup.py now?

Nice:

https://github.com/PyO3/setuptools-rust/blob/34c5f160a3a282ea568d0d285957edbdd927a9c7/pyproject.toml#L39

https://github.com/PyO3/setuptools-rust/blob/34c5f160a3a282ea568d0d285957edbdd927a9c7/setuptools_rust/setuptools_ext.py#L303-L314

Reply:
> > Does setuptools-rust not require a setup.py now?
> 
> Nice:
> 
> https://github.com/PyO3/setuptools-rust/blob/34c5f160a3a282ea568d0d285957edbdd927a9c7/pyproject.toml#L39
> 
> https://github.com/PyO3/setuptools-rust/blob/34c5f160a3a282ea568d0d285957edbdd927a9c7/setuptools_rust/setuptools_ext.py#L303-L314

I get what you mean more clearly now. Indeed.